System/Random.hs

```diff
@@ -1612,37 +1581,81 @@ unsignedBitmaskWithRejectionM genUniform range gen = go
         else pure x'
 {-# INLINE unsignedBitmaskWithRejectionM #-}
 
--- The global random number generator
+-------------------------------------------------------------------------------
+-- 'Uniform' instances for tuples
+-------------------------------------------------------------------------------
+
+instance (Uniform a, Uniform b) => Uniform (a, b) where
+  uniform g = do
+    a <- uniform g
+    b <- uniform g
+    return (a, b)
+
+instance (Uniform a, Uniform b, Uniform c) => Uniform (a, b, c) where
+  uniform g = do
+    a <- uniform g
+    b <- uniform g
+    c <- uniform g
+    return (a, b, c)
+
+instance (Uniform a, Uniform b, Uniform c, Uniform d) => Uniform (a, b, c, d) where
+  uniform g = do
```

Comment:
Maybe using applicative would be a bit more concise? It will avoid unnecessary bindings too
```
uniform g = (,,,) <$> uniform g <*> uniform g <*> uniform g <*> uniform g
```

Reply:
That's much better, thanks!